compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentGeneratorTest.kt

```diff
@@ -463,6 +464,72 @@ class ContributesSubcomponentGeneratorTest {
     }
   }
 
+  @Test fun `a contributed subcomponent can be replaced`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesSubcomponent
+
+      @ContributesSubcomponent(
+        scope = Any::class, 
+        parentScope = Unit::class
+      )
+      interface SubcomponentInterface1
+
+      @ContributesSubcomponent(
+        scope = Any::class, 
+        parentScope = Int::class,
+        replaces = [SubcomponentInterface1::class]
+      )
+      interface SubcomponentInterface2
+      """
+    ) {
+      assertThat(subcomponentInterface1.hintSubcomponent?.java).isEqualTo(subcomponentInterface1)
+      assertThat(subcomponentInterface1.hintSubcomponentParentScope).isEqualTo(Unit::class)
+
+      assertThat(subcomponentInterface2.hintSubcomponent?.java).isEqualTo(subcomponentInterface2)
+      assertThat(subcomponentInterface2.hintSubcomponentParentScope).isEqualTo(Int::class)
```

Comment:
At first glance, this test makes it look like both subcomponents are contributed successfully, rather than the first being replaced. I was expecting to see some sort of `assertFalse` style assert confirming that `SubcomponentInterface1` was not contributed. Is the reasoning that these second two asserts would fail if `SubcomponentInterface1` hadn't been replaced?

edit: I see that the test I was expecting is included in ContributesSubcomponentHandlerGeneratorTest.kt. It might be helpful to name this one as something like `a contributed subcomponent can be configured to replace another subcomponent`, if the intention is to make sure including `replaces = []` doesn't break anything.

Reply:
Very good suggestion!